app/src/lib/stores/pull-request-store.ts

```diff
@@ -326,22 +326,29 @@ export class PullRequestStore extends TypedBaseStore<GitHubRepository> {
 
     const table = this.pullRequestDatabase.pullRequests
     const prsToInsert = new Array<IPullRequest>()
-    let githubRepo: GitHubRepository | null = null
 
     for (const pr of pullRequestsFromAPI) {
-      // Once the repo is found on first try, no need to keep looking
-      if (githubRepo == null && pr.head.repo != null) {
-        githubRepo = await this.repositoryStore.upsertGitHubRepository(
-          repository.endpoint,
-          pr.head.repo
+      // `pr.head.repo` represents the source of the pull request. It might be
+      // a branch associated with the current repository, or a fork of the
+      // current repository.
+      //
+      // In cases where the user has removed the fork of the repository after
+      // opening a pull request, this can be `null`, and the app will not store
+      // this pull request.
+
+      if (pr.head.repo == null) {
+        log.debug(
+          `Unable to store pull request #${pr.number} for repository ${
+            repository.fullName
+          } as it has no head repository associated with it`
         )
+        continue
       }
 
-      if (githubRepo == null) {
```

Comment:
i understand why we fail silently if pr.head.repo is null/undefined and continue to fetch the githubRepo after with pr.head.repo as null/undefined but i'm not 100% sure why we don't return an error modal if the retrieved githubRepo is null or undefined. shouldn't we still do that, too?

Reply:
>  but i'm not 100% sure why we don't surface an error modal if the retrieved githubRepo is null or undefined. shouldn't we still do that, too?

I'm hiding this error away for a couple of reasons:

 - the app doesn't handle these PRs at all, so we'll need to address that if users want to checkout and merge them
 - this error would return after each fetch (when we refresh PRs in the background) - what can the user do about it beyond dismissing it?